pkg/util/slice/slice.go

```diff
@@ -56,20 +56,19 @@ func ContainsString(slice []string, s string, modifier func(s string) string) bo
 // RemoveString returns a newly created []string that contains all items from slice that
 // are not equal to s and modifier(s) in case modifier func is provided.
 func RemoveString(slice []string, s string, modifier func(s string) string) []string {
-	newSlice := make([]string, 0)
+	index := 0
 	for _, item := range slice {
 		if item == s {
 			continue
 		}
 		if modifier != nil && modifier(item) == s {
 			continue
 		}
-		newSlice = append(newSlice, item)
+		slice[index] = item
+		index++
 	}
-	if len(newSlice) == 0 {
-		// Sanitize for unit tests so we don't need to distinguish empty array
-		// and nil.
-		newSlice = nil
+	if index == 0 {
+		return nil
 	}
-	return newSlice
+	return slice[:index]
```

Comment:
I'm not sure doing this will be a good idea. While we might get small performance gains (to be confirmed by benchmarks), note the documentation comment of the function: `RemoveString returns a newly created []string`, emphasis on _newly_. Slicing an existing slice will still point to the same underlying array. If callers of the function rely on this documented property, then it can lead to weird behaviours because elements modified in the returned slice will also modify elements of the original slice.